consensus/spos/bls/subroundSignature.go

```diff
@@ -379,58 +383,84 @@ func (sr *subroundSignature) remainingTime() time.Duration {
 		}
 
-		isCurrentManagedKeyLeader := idx == spos.IndexOfLeaderInConsensusGroup
-		// TODO[cleanup cns finality]: update the check
-		// with the equivalent messages feature on, signatures from all managed keys must be broadcast, as the aggregation is done by any participant
-		shouldBroadcastSignatureShare := (!isCurrentNodeMultiKeyLeader && !isFlagActive) ||
-			(!isCurrentManagedKeyLeader && isFlagActive)
-		if shouldBroadcastSignatureShare {
-			ok := sr.createAndSendSignatureMessage(signatureShare, pkBytes)
-			if !ok {
-				return false
+		go func(idx int, pk string) {
+
+			signatureSent := sr.sendSignatureForManagedKey(idx, pk)
+			if signatureSent {
+				atomic.AddInt32(&numMultiKeysSignaturesSent, 1)
+			} else {
+				sentSigForAllKeys.SetValue(false)
 			}
+			wg.Done()
 
-			numMultiKeysSignaturesSent++
-		}
-		// with the equivalent messages feature on, the leader signature is sent on subroundBlock, thus we should update its status here as well
-		sr.sentSignatureTracker.SignatureSent(pkBytes)
+		}(idx, pk)
+	}
+	wg.Wait()
+
+	if numMultiKeysSignaturesSent > 0 {
+		log.Debug("step 2: multi keys signatures have been sent", "num", numMultiKeysSignaturesSent)
+	}
+
+	return sentSigForAllKeys.IsSet()
+}
+
+func (sr *subroundSignature) sendSignatureForManagedKey(idx int, pk string) bool {
+	isCurrentNodeMultiKeyLeader := sr.IsMultiKeyLeaderInCurrentRound()
+	isFlagActive := sr.EnableEpochsHandler().IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, sr.Header.GetEpoch())
+
+	pkBytes := []byte(pk)
+
+	signatureShare, err := sr.SigningHandler().CreateSignatureShareForPublicKey(
+		sr.GetData(),
+		uint16(idx),
+		sr.Header.GetEpoch(),
+		pkBytes,
+	)
+	if err != nil {
+		log.Debug("doSignatureJobForManagedKeys.CreateSignatureShareForPublicKey", "error", err.Error())
+		return false
+	}
+
+	isCurrentManagedKeyLeader := idx == spos.IndexOfLeaderInConsensusGroup
+	// TODO[cleanup cns finality]: update the check
+	// with the equivalent messages feature on, signatures from all managed keys must be broadcast, as the aggregation is done by any participant
+	shouldBroadcastSignatureShare := (!isCurrentNodeMultiKeyLeader && !isFlagActive) ||
+		(!isCurrentManagedKeyLeader && isFlagActive)
+	if shouldBroadcastSignatureShare {
+		ok := sr.createAndSendSignatureMessage(signatureShare, pkBytes)
 
-		shouldWaitForAllSigsAsync := isCurrentManagedKeyLeader && !isFlagActive
-		ok := sr.completeSignatureSubRound(pk, shouldWaitForAllSigsAsync)
 		if !ok {
 			return false
 		}
-	}
 
-	if numMultiKeysSignaturesSent > 0 {
-		log.Debug("step 2: multi keys signatures have been sent", "num", numMultiKeysSignaturesSent)
 	}
+	// with the equivalent messages feature on, the leader signature is sent on subroundBlock, thus we should update its status here as well
+	sr.sentSignatureTracker.SignatureSent(pkBytes)
 
-	return true
+	shouldWaitForAllSigsAsync := isCurrentManagedKeyLeader && !isFlagActive
+	ok := sr.completeSignatureSubRound(pk, shouldWaitForAllSigsAsync)
```

Comment:
you can directly return here instead of setting the variable ok and then returning it

return sr.completeSignatureSubRound(pk, shouldWaitForAllSigsAsync)

Reply:
updated